calling-apis/intro-to-asynchronous-programming.md

```diff
@@ -0,0 +1,136 @@
+
+1. call actions/processes outside of program flow, so that the program doesn't stop for a result
+2. define what happens when the result comes back
+
+Asynchronous programming allows us to program behavior where some code _does not wait_ for other code to finish.
+
+To set up asynchronous programming, we will need to consider initially:
+
+- Which specific technologies (tools and environment) to use that support running asynchronous code
+- While or after an asynchronous call, what should happen, and all the different cases of success and failure
+
+## Writing Asynchronous Code
+
+To write good asynchronous code, we will have to determine and write the following things:
+
+| Thing to Determine                                                                                                                  | Food Delivery Metaphor                                                                                                                                                                            |
+| ----------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| What is the asynchronous function we are invoking?                                                                                  | How do we order the food?                                                                                                                                                                         |
+| What do we do after the asynchronous call finishes?                                                                                 | What do we do after we receive the food delivery?                                                                                                                                                 |
+| What do we do if the async call doesn't finish successfully?                                                                        | What do we do if the food never arrives? What do we do if the restaurant calls back and says that they are out of food? What do we do if the restaurant calls back and says they lost your order? |
+| How can we ensure that every other part of the program runs correctly, without bugs, even if the asynchronous call hasn't finished? | How do we make sure that Mica isn't too hungry and orders another dinner? If Mica's friend joins them and they are hungry and wants to eat, what does Mica do?                                    |
+
+## Summary
+
+Typically, we write a lot of code that runs in a very specific order (synchronous). However, certain technologies allow us to write asynchronous code, and we'll see plenty in JavaScript.
```

Comment:
restate plenty of what. the technologies reference goes by quickly so I was left needing to go back and try to figure out what plenty referred to.
```suggestion
Typically, we write a lot of code that runs in a very specific order (synchronous). However, certain technologies allow us to write asynchronous code, and we'll see plenty of these technologies in JavaScript.
```